uefi/src/allocator.rs

```diff
@@ -169,3 +169,17 @@ unsafe impl GlobalAlloc for Allocator {
         }
     }
 }
+
+#[cfg(feature = "unstable")]
+unsafe impl core::alloc::Allocator for Allocator {
+    fn allocate(&self, layout: Layout) -> Result<NonNull<[u8]>, core::alloc::AllocError> {
+        let ptr = unsafe { <Allocator as GlobalAlloc>::alloc(self, layout) };
```

Comment:
Do we need to do something here to handle ZSTs? I'm not familiar with the details of the `Allocator` trait, but I noticed this in the [documentation](https://doc.rust-lang.org/std/alloc/trait.Allocator.html):

> In contrast to [`GlobalAlloc`](https://doc.rust-lang.org/std/alloc/trait.GlobalAlloc.html), `Allocator` allows zero-sized allocations. If an underlying allocator does not support this (like jemalloc) or responds by returning a null pointer (such as `libc::malloc`), this must be caught by the implementation.

Reply:
Yes, I'm just not sure how to handle it :D but I can figure it out, I guess

Reply:
Nice use case but you're right. I added it in a dedicated commit